From 3730f52d69b5627b8f65d1c410cd3dd1da98e93f Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Tue, 26 Aug 2014 12:49:38 -0400
Subject: [tools] cpupower: drop negativity check on unsigned value

Message-id: <1409057382-26432-28-git-send-email-prarit@redhat.com>
Patchwork-id: 89095
O-Subject: [RHEL7.1 PATCH BZ 1133893 27/31] PM / tools: cpupower: drop negativity check on unsigned value
Bugzilla: 1133893
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1133893

commit 788606cb8864765bb16be23d368725df84d0e3cc
Author: Andrey Utkin <andrey.krieger.utkin@gmail.com>
Date:   Sat Jul 19 22:14:04 2014 +0300

    PM / tools: cpupower: drop negativity check on unsigned value

    Link: https://bugzilla.kernel.org/show_bug.cgi?id=80621
    Reported-by: David Binderman <dcb314@hotmail.com>
    Signed-off-by: Andrey Utkin <andrey.krieger.utkin@gmail.com>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Tony Camuso <tcamuso@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/power/cpupower/utils/helpers/sysfs.c b/tools/power/cpupower/utils/helpers/sysfs.c
index 851c7a1..09afe5d 100644
--- a/tools/power/cpupower/utils/helpers/sysfs.c
+++ b/tools/power/cpupower/utils/helpers/sysfs.c
@@ -81,7 +81,7 @@ int sysfs_is_cpu_online(unsigned int cpu)
  close(fd);
 
  value = strtoull(linebuf, &endp, 0);
- if (value > 1 || value < 0)
+ if (value > 1)
   return -EINVAL;
 
  return value;
-- 
1.7.1